polkadot/node/core/approval-voting/src/lib.rs

```diff
@@ -182,6 +186,8 @@ struct MetricsInner {
 	time_recover_and_approve: prometheus::Histogram,
 	candidate_signatures_requests_total: prometheus::Counter<prometheus::U64>,
 	unapproved_candidates_in_unfinalized_chain: prometheus::Gauge<prometheus::U64>,
+	// The time it takes in each stage to gather enough assignments.
+	assignments_gathering_time_by_stage: prometheus::HistogramVec,
```

Comment:
Stage is quite confusing here, let's describe exactly what stage is.

Reply:
Yeah, I was split between stage, round and depth is any of it more explicit for you ?